src/js/index.js

```diff
@@ -28,58 +51,126 @@ function App() {
+    this.menu[this.currentCategory] = await MenuApi.getAllMenuByCategory(
+      this.currentCategory
     )
 
-    updateMenuCount()
+    $('#menu-list').innerHTML = template(this.menu[this.currentCategory])
 
-    $('#espresso-menu-name').value = ''
+    updateMenuCount()
   }
 
   const updateMenuCount = () => {
-    const menuCount = $('#espresso-menu-list').querySelectorAll('li').length
+    const menuCount = this.menu[this.currentCategory].length
     $('.menu-count').innerText = `총 ${menuCount} 개`
   }
 
-  const updateMenuName = (e) => {
+  const addMenuName = async () => {
+    if ($('#menu-name').value.trim() === '') {
+      alert('값을 입력해주세요.')
+
+      return
+    }
+
+    const duplicatedItem = this.menu[this.currentCategory].find(
+      (menuItem) => menuItem.name === $('#menu-name').value
+    )
+    if (duplicatedItem) {
+      alert('이미 등록된 메뉴입니다. 다시 입력해주세요.')
+      $('#menu-name').value = ''
+
+      return
+    }
+
+    const menuName = $('#menu-name').value
+    await MenuApi.createMenu(this.currentCategory, menuName)
+
+    render()
+    $('#menu-name').value = ''
+  }
+
+  const updateMenuName = async (e) => {
     const $menuName = e.target.closest('li').querySelector('.menu-name')
     const updatedMenuName = prompt('메뉴명을 수정하세요', $menuName.innerText)
-    if (updatedMenuName) {
-      $menuName.innerText = updatedMenuName
+    if (!updatedMenuName) {
+      return
     }
+
+    await MenuApi.updateMenu(this.currentCategory, updatedMenuName, $menuId(e))
+
+    render()
   }
 
-  const removeMenuName = (e) => {
+  const removeMenuName = async (e) => {
     if (confirm('정말 삭제하시겠습니까?')) {
-      e.target.closest('li').remove()
+      await MenuApi.deleteMenu(this.currentCategory, $menuId(e))
 
-      updateMenuCount()
+      render()
     }
   }
 
-  $('#espresso-menu-submit-button').addEventListener('click', addMenuName)
+  const soldOutMenu = async (e) => {
+    await MenuApi.toggleSoldOutMenu(this.currentCategory, $menuId(e))
+
+    render()
+  }
 
-  $('#espresso-menu-name').addEventListener('keypress', (e) => {
+  const enterMenuName = (e) => {
     if (e.key !== 'Enter') {
       return
     }
 
     addMenuName()
-  })
+  }
 
-  $('#espresso-menu-list').addEventListener('click', (e) => {
+  const handleMenuList = (e) => {
     if (e.target.classList.contains('menu-edit-button')) {
       updateMenuName(e)
+
+      return
     }
 
     if (e.target.classList.contains('menu-remove-button')) {
       removeMenuName(e)
+
+      return
     }
-  })
+
+    if (e.target.classList.contains('menu-sold-out-button')) {
+      soldOutMenu(e)
+
+      return
+    }
+  }
+
+  const changeCategory = (e) => {
+    const isCategoryButton = e.target.classList.contains('cafe-category-name')
+    if (isCategoryButton) {
+      const categoryName = e.target.dataset.categoryName
+      this.currentCategory = categoryName
+      $('#category-title').innerText = `${e.target.innerText} 메뉴 관리`
+
+      render()
+    }
```

Comment:
```suggestion
    const isCategoryButton = e.target.classList.contains('cafe-category-name')
    if (!isCategoryButton) { 
      return
    }
    const categoryName = e.target.dataset.categoryName
    this.currentCategory = categoryName
    $('#category-title').innerText = `${e.target.innerText} 메뉴 관리`
    render()
    }
```
이렇게 early return을 해주시면 로직이 더 깔끔할 것 같아요~!